deltacat/utils/rangedictionary.py

```diff
@@ -0,0 +1,92 @@
+from collections import OrderedDict
+from typing import Any, Optional, Union
+
+
+class IntegerRangeDict(OrderedDict):
```

Comment:
Is `SortedDict` more accurate name? Also, there seems to be a [SortedDict](https://www.geeksforgeeks.org/python-sorted-containers-an-introduction/) which does exactly what you are trying to do. Can we leverage the same?